core/src/main/scala-3/spire/syntax/macros/cforMacros.scala

```diff
@@ -20,11 +20,34 @@ import collection.immutable.NumericRange
 
 import spire.syntax.fastFor.{RangeElem, RangeLike}
 
-inline def fastForInline[R](init: R, test: R => Boolean, next: R => R, body: R => Unit): Unit =
-  var index = init
-  while test(index) do
-    body(index)
-    index = next(index)
+def fastForImpl[R: Type](init: Expr[R], test: Expr[R => Boolean], next: Expr[R => R], body: Expr[R => Unit])(using
+  Quotes
+): Expr[Unit] =
+
+  def fastCode =
+    for
+      testOpt <- Util.tryOpt(test)
+      nextOpt <- Util.tryOpt(next)
+      bodyOpt <- Util.tryOpt(body)
+    yield '{
```

Comment:
Forgive my ignorance, but is there a reason we need `tryOpt` to succeed for all three of these together? Would it be possible to have a "hybrid" code which uses fast code for some of these and slow code for the others?

Reply:
this was just my first try :) I agree it would be nice to have the ability to only have one deoptimised argument - I'll see what can be done

Reply:
I have implemented the ability to optimise function arguments independently of each other, and included it on fastForRange, allowing it to match the behaviour of cforRange